packages/core/http/core-http-router-server-internal/src/request.ts

```diff
@@ -222,6 +223,13 @@ export class CoreKibanaRequest<
       options,
     };
   }
+  /** infer route access from path if not declared */
+  private getAccess(request: RawRequest): 'internal' | 'public' {
+    return (
+      ((request.route?.settings as RouteOptions)?.app as KibanaRouteOptions)?.access ??
+      (request.path.includes('internal') ? 'internal' : 'public')
```

Comment:
Um, I thought I had changed the string match to `startsWith`. Not sure where that commit got lost.